libs/react-helpers/src/hooks/use-data-provider.ts

```diff
@@ -27,7 +33,9 @@ export function useDataProvider<Data, Delta>(
     [update]
   );
   useEffect(() => {
-    return dataProvider(callback, client, variables);
+    const { unsubscribe, flush } = dataProvider(callback, client, variables);
+    flushRef.current = flush;
+    return unsubscribe;
   }, [client, initialized, dataProvider, callback, variables]);
-  return { data, loading, error };
+  return { data, loading, error, flush };
```

Comment:
flush can be used to get updated data from provider even if component passes update function